crates/ruff/src/linter.rs

```diff
@@ -63,7 +63,7 @@ pub struct FixerResult<'a> {
     /// The result returned by the linter, after applying any fixes.
     pub result: LinterResult<(Vec<Message>, Option<ImportMap>)>,
     /// The resulting source code, after applying any fixes.
-    pub transformed: Cow<'a, str>,
+    pub transformed: Cow<'a, SourceKind>,
```

Comment:
The change here is that the linter now returns the transformed `SourceKind` which aligns the handling of Notebooks with regular python files in the sense that it doesn't mutate the source file in place.